### PR TITLE
Stub Video.videoWidth and videoHeight and port Video to ActionScript

### PR DESCRIPTION
**Pull request by Lord-McSweeney**
Progresses #8006. 320 and 240 are the default width and height.

### CODE REVIEW

**Comment by danielhjacobs** [comment]
I suspect moving this file to AS at the same time may be better.

**Comment by Lord-McSweeney** [comment]
I agree. Marking as draft until #9385 is merged.

File: core/src/avm2/globals/flash/media/Video.as [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package flash.media {
+    public final class Video {
```
Comment (adrian17) [inline core/src/avm2/globals/flash/media/Video.as:2]: it's missing a DisplayObject inheritance; also it's not `final`

File: core/src/avm2/globals/flash/media/Video.as [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        private var _videoWidth: int;
+        private var _videoHeight: int;
+
+        public function Video(videoWidth:int = 320, videoHeight:int = 240) {
```
Comment (adrian17) [inline core/src/avm2/globals/flash/media/Video.as:6]: let's add a stub mention to the constructor

**Review by adrian17 — COMMENTED** [review]
(Note to whoever merges: let's not forget to auto-squash)

File: core/src/avm2/globals/flash/media/Video.as (hunk 2 above)
Comment (danielhjacobs) [reply core/src/avm2/globals/flash/media/Video.as:9]: I think Adrian meant like https://github.com/ruffle-rs/ruffle/blob/master/core/src/avm2/globals/flash/utils/Dictionary.as#L10

Comment (Lord-McSweeney) [reply core/src/avm2/globals/flash/media/Video.as:9]: `Video` doesn't actually do anything itself; it only serves as a container to be filled by `Video.attachCamera` or `Video.attachNetStream`. Adding a stub warning will unnecessarily warn about the Video stubbed on content that doesn't use `attachNetStream`, in which case the `Video` container is functioning correctly.

File: core/src/avm2/globals/flash/media/Video.as (hunk 1 above)
Comment (relrelb) [reply core/src/avm2/globals/flash/media/Video.as]: Agree on `DisplayObject ` inheritance, but actually `Video` is not declared as `final` as per https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/Video.html.

**Comment by danielhjacobs** [comment]
Did you mean to close this and force-push and remove all commits?

File: core/src/avm2/globals/flash/media/Video.as (hunk 1 above)
Comment (adrian17) [reply core/src/avm2/globals/flash/media/Video.as]: >  actually Video is not declared as final

yeah, that's what I meant :) That comment was from the version of this PR that had `final`.

**Pull request closed without merging**